src/test/codegen/noalias-rwlockreadguard.rs

```diff
@@ -0,0 +1,15 @@
+// compile-flags: -O -C no-prepopulate-passes -Z mutable-noalias=yes
+
+#![crate_type = "lib"]
+
+use std::cell::UnsafeCell;
+use std::sync::RwLockReadGuard;
+
+// Make sure that `RwLockReadGuard` does not get a `noalias` attribute, because
+// the `UnsafeCell` might alias writes after it is dropped.
+
+// CHECK-LABEL: @maybe_aliased(
+// CHECK-NOT: noalias
+// CHECK-SAME: %_data
+#[no_mangle]
+pub unsafe fn maybe_aliased(_: RwLockReadGuard<'_, i32>, _data: &UnsafeCell<i32>) {}
```

Comment:
Can you change the second parameter to `&RwLock<i32>`? That makes a more realistic example of aliasing, whereas only the standard library itself ever has access to the underlying `UnsafeCell` that would alias a guard.

Reply:
[048a80](https://github.com/rust-lang/rust/pull/96820/commits/048a80140b82471b671e7ae17f6dce2bb604f473)